mne/decoding/tests/test_transformer.py

```diff
@@ -229,3 +229,49 @@ def test_unsupervised_spatial_filter():
     usf = UnsupervisedSpatialFilter(PCA(4), average=True)
     usf.fit_transform(X)
     assert_raises(ValueError, UnsupervisedSpatialFilter, PCA(4), 2)
+
+
+def test_temporal_filterer():
+    """Test methods of TemporalFilter."""
+    X = np.random.rand(10, 5, 2)
+
+    # Add tests for different combinations of l_freq and h_freq
+    filt = TemporalFilter(l_freq=5, h_freq=15, sfreq=100,
+                          filter_length='auto', l_trans_bandwidth=None,
+                          h_trans_bandwidth=None)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        result = filt.fit_transform(X)
+        assert_true(X.shape == result.shape)
+        assert_array_equal(filt.fit(X).transform(X), result)
+
+    filt = TemporalFilter(l_freq=None, h_freq=15, sfreq=100,
+                          filter_length='auto', l_trans_bandwidth=None,
+                          h_trans_bandwidth=None)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        X = filt.fit_transform(X)
```

Comment:
don't call you transformed value `X`, but `Xt`, same everywhere. Else you'll filter the signal many times, and can't really compare the results across different transforms.